### PR TITLE
[FIX] #93564 sale_timesheet: profitable project report handle partial refund

### LINKED ISSUE #93564: Project overview ignore partial refund invoices
## Impacted versions:

Tested on v14.0: https://16444287-14-0-all.runbot89.odoo.com/

## Steps to reproduce:

* enable Analytic Accounting (accounting > settings>
* create a billable project (project)
* create an invoice and link to the analytic account (ie 33 items at $10/unit => $ 330 HT) and post it (Accounting > customer invoice)
* generate a partial refund and change quantities (ie: from 33 -> 3 units => $30HT)
* go to account analytic lines you get 2 lines +330 - 30 => 300 (which if good) (Invoicing > Configuration > Analytics Item)
 
image

* go to the project overview (project) Total Profitability is 0

image

## Current behavior:

partial refund analytic lines with changed quantity are ignored in the profitability report $0 display as total

## Expected behavior:

Take care of partial profitability and display proper result (in this example $300)

## Analysis

The project overview is based on `project.profitability.report` model from `sale_timesheet`. This model is based on a SQL view that ignore refund lines positive as negative, if we don't want to display refunds as expense we should probably compute the difference between both line as revenue ?!

### PR DESCRIPTION
**Pull request by petrus-v**
This fix profitability total report while using partial refund on customer/vendor invoices.
    
To reproduce this issue:
    * enable Analytic Accounting
    * create a billable project
    * create an invoice and link to the analytic account and post it (ie: 33 items at $ 10/unit $ 330 HT)
    * generate a partial refund and change quantities (ie: from 33 to 3 units => $ 30HT)
    * check account analytic lines are correct (get 2 lines +330 - 30 => $ 300)
    * go to the project overview and check other revenues and Total Profitability
    
Before this commit total is $ 0
After this commit total is invoiced amount - refund amount ($ 300)

More details in the related issue: #93564

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]

**Review by IT-Ideas — APPROVED** [review]
Thanks for the fix @petrus-v !
@robodoo r+

File: addons/sale_timesheet/report/project_profitability_report_analysis.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def init(self):
                                     AAL.so_line AS sale_line_id,
                                     0.0 AS timesheet_unit_amount,
                                     0.0 AS timesheet_cost,
-                                    AAL.amount AS other_revenues,
+                                    AAL.amount + COALESCE(AAL_RINV.amount, 0) AS other_revenues,
```
Comment (IT-Ideas) [inline addons/sale_timesheet/report/project_profitability_report_analysis.py:167]: Could you rename your commit like this?
```
[FIX] sale_timesheet: profitable project report handle partial refund

This fix profitability total report while using partial refund on customer/vendor invoices.

To reproduce this issue:
* enable Analytic Accounting
* create a billable project
* create an invoice and link to the analytic account and post it (ie: 33 items at $ 10/unit $ 330 HT)
* generate a partial refund and change quantities (ie: from 33 to 3 units => $ 30HT)
* check account analytic lines are correct (get 2 lines +330 - 30 => $ 300)
* go to the project overview and check other revenues and Total Profitability

Before this commit total is $ 0
After this commit total is invoiced amount - refund amount ($ 300)

Fixes #93564
```

File: doc/cla/individual/petrus-v.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+France, 2021-03-22
```
Comment (IT-Ideas) [inline doc/cla/individual/petrus-v.md:1]: Can you rename your commit title like
```
[CLA] Sign Odoo CLA for petrus-v
```

**Review by IT-Ideas — COMMENTED** [review]
Can you just rename your commits as I mentioned in the review?
Thanks 🙂 
Once done I'll launch the merging process.

File: addons/sale_timesheet/report/project_profitability_report_analysis.py (hunk 1 above)
Comment (petrus-v) [reply addons/sale_timesheet/report/project_profitability_report_analysis.py:167]: done !

**Comment by IT-Ideas** [comment]
Thanks 🙂 
@robodoo r+

**Comment by IT-Ideas** [comment]
@robodoo rebase-ff

**Pull request closed without merging**